src/legacy/core_plugins/kibana/public/visualize/editor/editor.js

```diff
@@ -343,13 +343,33 @@ function VisEditor(
     $scope.$watchMulti([
       'searchSource.getField("index")',
       'vis.type.options.showTimePicker',
-    ], function ([index, requiresTimePicker]) {
+      'vis.type.options.showQueryBar',
+    ], function ([index, requiresTimePicker, requiresQueryBar]) {
       const showTimeFilter = Boolean((!index || index.timeFieldName) && requiresTimePicker);
 
-      if (showTimeFilter) {
+      if (showTimeFilter && requiresQueryBar) {
```

Comment:
Nit: These conditions could be simplified, not sure if you just wanted to be explicit about the four different conditions here. In particular the first and third conditions are almost identical and could be replaced with something like

```javascript
      if (requiresQueryBar) {
        timefilter.disableTimeRangeSelector();
        timefilter.disableAutoRefreshSelector();
        $scope.enableQueryBarTimeRangeSelector = true;
        $scope.showAutoRefreshOnlyInQueryBar = !showTimeFilter;
      }
      else if (showTimeFilter) {
        timefilter.enableTimeRangeSelector();
        timefilter.enableAutoRefreshSelector();
        $scope.enableQueryBarTimeRangeSelector = false;
        $scope.showAutoRefreshOnlyInQueryBar = false;
      }
      else {
        timefilter.disableTimeRangeSelector();
        timefilter.enableAutoRefreshSelector();
        $scope.enableQueryBarTimeRangeSelector = false;
        $scope.showAutoRefreshOnlyInQueryBar = false;
      }
```

Reply:
Good point. As I was trying to think through all the combinations it was just easier to have a separate `else` for each one. I've simplified it as you suggested. Feel free to make these sort of suggestions using the GH suggestion feature and I'll merge them right in 😉